crates/handlers/src/views/recovery/progress.rs

```diff
@@ -130,14 +132,25 @@ pub(crate) async fn post(
     // Verify the CSRF token
     let () = cookie_jar.verify_form(&clock, form)?;
 
+    // Check the rate limit if we are about to process the form
+    if let Err(e) = limiter.check_account_recovery(requester, &recovery_session.email) {
+        tracing::warn!(error = &e as &dyn std::error::Error);
```

Comment:
I really wish tracing had a sigil (like `?` or `%`) to report as `&dyn Error` :(

Reply:
https://github.com/tokio-rs/tracing/issues/1308